programs/drift/src/controller/amm.rs

```diff
@@ -542,7 +543,36 @@ pub fn update_pool_balances(
     let pnl_to_settle_with_user = if user_unsettled_pnl > 0 {
         min(user_unsettled_pnl, pnl_pool_token_amount.cast::<i128>()?)
     } else {
-        user_unsettled_pnl
+        // dont settle negative pnl to spot borrows when utilization is high
+        let utilization =
```

Comment:
why this 80% utilization threshold instead of calculating borrow liquidity based on borrow guardrails?

Reply:
feel like we can reuse `get_max_fill_amounts_for_market` to figure max negative pnl that can be settled